### PR TITLE
feat(components/login/passwordReset): Add more semantic errors

### PR DESCRIPTION
**Pull request by oegea**
# Context

We're implementing the `login/passwordReset` component. This component will allow any web-app to render the password reset interface in a fully functional way, in order to adopt the cross login system ensuring the best developer experience possible.

# Changes

Stop using standard errors and wrap them on more semantic error class.

### CODE REVIEW

**Review by davidmartin84 — APPROVED** [review]

File: components/login/passwordReset/src/domain/password/Repositories/HTTPPasswordRepository.js [hunk 1]
```diff
@@ -3,10 +3,11 @@ import {inlineError} from '@s-ui/decorators'
 import {PasswordRepository} from './PasswordRepository.js'
 
 export class HTTPPasswordRepository extends PasswordRepository {
-  constructor({config, fetcher}) {
+  constructor({config, fetcher, passwordError}) {
     super()
     this._config = config
     this._fetcher = fetcher
+    this._passwordError = passwordError
```
Comment (quinwacca) [inline components/login/passwordReset/src/domain/password/Repositories/HTTPPasswordRepository.js:10]: what do you think about rename it this way? ⬇️ 
```suggestion
    this._passwordErrorFactory = passwordErrorFactory
```

File: components/login/passwordReset/src/domain/password/Repositories/HTTPPasswordRepository.js [hunk 2]
```diff
@@ -18,7 +19,7 @@ export class HTTPPasswordRepository extends PasswordRepository {
       .post({path, params: {password, token}})
       .then(() => true)
       .catch(() => {
-        throw new Error('Unhandled error ocurred')
+        throw this._passwordError('Unhandled error ocurred')
```
Comment (jordevo) [inline components/login/passwordReset/src/domain/password/Repositories/HTTPPasswordRepository.js:22]:
```suggestion
        throw this._passwordError('Unhandled error occurred')
```

🤓

**Review by jordevo — APPROVED** [review]

**Review by quinwacca — APPROVED** [review]

File: components/login/passwordReset/src/domain/password/Repositories/HTTPPasswordRepository.js (hunk 1 above)
Comment (oegea) [reply components/login/passwordReset/src/domain/password/Repositories/HTTPPasswordRepository.js:10]: I don't agree with the proposal 😋 
IMHO including the factory word in the property name may induce to  think that the property contains a reference to the fully factory class. And it only contains the specific `passwordError` initializer.

File: components/login/passwordReset/src/domain/password/Repositories/HTTPPasswordRepository.js (hunk 2 above)
Comment (oegea) [reply components/login/passwordReset/src/domain/password/Repositories/HTTPPasswordRepository.js:22]: Thank you!

File: components/login/passwordReset/src/domain/test/password/ChangePasswordUseCaseSpec.js [hunk 3]
```diff
@@ -42,7 +42,9 @@ describe('[Domain] ChangePasswordUseCase', () => {
 
     const [error, result] = await useCase.execute(requestBody)
     expect(error).to.not.be.null
-    expect(error.toString()).to.eql('Error: Unhandled error ocurred')
+    expect(error.toString()).to.eql(
+      'Error: [PasswordError] Unhandled error ocurred'
+    )
```
Comment (AgonisticKatai) [inline components/login/passwordReset/src/domain/test/password/ChangePasswordUseCaseSpec.js:47]: Could be interesting to verify this error is the expected instance of `PasswordError`

**Review by AgonisticKatai — APPROVED** [review]

File: components/login/passwordReset/src/domain/test/password/ChangePasswordUseCaseSpec.js (hunk 3 above)
Comment (oegea) [reply components/login/passwordReset/src/domain/test/password/ChangePasswordUseCaseSpec.js:47]: Thank u!

**Pull request merged**